src/main/webapp/app/entities/exercise-filter.model.ts

```diff
@@ -0,0 +1,36 @@
+import { Exercise } from 'app/entities/exercise.model';
+import { ExerciseCategory } from 'app/entities/exercise-category.model';
+
+export class ExerciseFilter {
+    exerciseNameSearch: string;
+    exerciseCategorySearch: string;
+    exerciseTypeSearch: string;
+
+    constructor(exerciseNameSearch = '', exerciseCategorySearch = '', exerciseTypeSearch = 'all') {
+        this.exerciseNameSearch = exerciseNameSearch.toLowerCase();
+        this.exerciseCategorySearch = exerciseCategorySearch.toLowerCase();
+        this.exerciseTypeSearch = exerciseTypeSearch;
+    }
+
+    /**
+     * Check if the exercise filter is in its default state (no exercise will be filtered by it)
+     */
+    isEmpty(): boolean {
+        return this.exerciseNameSearch === '' && this.exerciseCategorySearch === '' && this.exerciseTypeSearch === 'all';
+    }
+
+    /**
+     * Checks if an exercise should be included by this filter
+     * @param exercise The exercise to check
+     */
+    includeExercise(exercise: Exercise): boolean {
```

Comment:
I think having prefix `is` or `does` helps understand it's a predicate and not an action of "including" the exercise somewhere.
Or maybe just add `s` such that it is named `includes` so it is similar with the collection method you were using before. Or maybe even `matchesExercise` as `match` word is used a lot in your method and you have already a `matchesTag` method.